contrib/scalapblib/src/ScalaPBWorker.scala

```diff
@@ -25,17 +25,18 @@ class ScalaPBWorker {
 
         val instance = new ScalaPBWorkerApi {
           override def compileScalaPB(
-              source: File,
+              root: File,
+              sources: Seq[File],
               scalaPBOptions: String,
               generatedDirectory: File,
               otherArgs: Seq[String]
-          ): Unit = {
+            ): Unit = {
             val opts = if (scalaPBOptions.isEmpty) "" else scalaPBOptions + ":"
             val args = otherArgs ++ Seq(
               s"--scala_out=${opts}${generatedDirectory.getCanonicalPath}",
-              s"--proto_path=${source.getParentFile.getCanonicalPath}",
-              source.getCanonicalPath
-            )
+              s"--proto_path=${root.getCanonicalPath}"
+            ) ++ sources.map(_.getCanonicalPath)
+            println(args)
```

Comment:
Looks like a debugging left over. If not, please use an appropriate method or stream of `ctx.log`, e.g. `debug` for debug info.

Reply:
Good catch, I've made it a debug log i think it is useful to see the exact command parameters passed to the generator then you can run always run it manually to find out what is 'wrong'.